### PR TITLE
refactor(web-client): renovate session state and services

### PR DESCRIPTION
**Pull request by PiDelport**
This renovates the session state management and services a fair bit, in preparation for merging the ASA and XRPL work:

1. Separate `AlgodService` from `EnclaveService`, and define their responsibilities more clearly:

   - `EnclaveService` provides for communication the wallet enclave server
   - `AlgodService` provides for communication with the Algod node
 
2. Rework `SessionState` to represent domain data faithfully. Everything should now be stored in a way that avoids semantic conflicts; in particular:

   - Remove the global `balance` field, in favour of representing specific balances
   - Remove uses of `walletId` as Algorand address
   - Remove the transient `transactionId` field, in favour of local state

3. Move management of session state and operations into session-attached services (following Akita's pattern):

   - `SessionService` for core state and operations associated with the enclave (via `EnclaveService`)
   - `SessionAlgorandService` for state and operations associated with the Algorand ledger (via `AlgodService`)

### Related PRs

- This should precede #131 and #115

### CODE REVIEW

File: web-client/src/app/services/algod.service.ts [hunk 1]
```diff
@@ -1,8 +1,83 @@
...
+  /** Combine {@link submitSignedTransaction} and {@link waitForTransactionConfirmation}. */
+  async submitAndConfirmTransaction(
+    signedTxn: Uint8Array
+  ): Promise<TransactionConfirmation> {
+    const { txId } = await this.submitSignedTransaction(signedTxn);
+    return await this.waitForTransactionConfirmation(txId);
+  }
 }
+
+const getAlgodClientFromEnvironment = () => {
```
[the remark this answers is not in the document]
Comment (PiDelport) [reply web-client/src/app/services/algod.service.ts:79]: This is probably better, yeah, though I'm happy to leave this as a `TODO` for now?

**Review by longtomjr — APPROVED** [review]
Looks good. Thanks a lot!

File: web-client/src/app/services/algosdk.utils.ts [hunk 2]
```diff
@@ -102,3 +103,42 @@ const checkString = (value: unknown): string => {
...
+// Docs: https://developer.algorand.org/docs/reference/rest-apis/algod/v2/#account
+export type AccountData = {
+  address: string;
+  amount: number;
+  assets?: AssetHolding[];
+};
+
+// Docs: https://developer.algorand.org/docs/rest-apis/algod/v2/#assetholding
+export type AssetHolding = {
+  amount: number;
```
[the remark this answers is not in the document]
Comment (PiDelport) [reply web-client/src/app/services/algosdk.utils.ts:164]: I began looking at doing something for this PR, but it's a significant rabbithole… I filed this wishlist issue instead:

- https://github.com/ntls-io/nautilus-wallet/issues/165

In the interim, I changed our client construction to specify `IntDecoding.SAFE`, and added some comments about it.

File: web-client/src/app/state/session-algorand.service.ts [hunk 3]
```diff
@@ -1,8 +1,95 @@
...
+    const algorandAccountData = await this.algodService.getAccountData(
+      wallet.algorand_address_base32
+    );
+    this.sessionStore.update({ algorandAccountData });
+  }
+
+  /**
+   * Send funds to another account.
+   */
+  async sendFunds(
```
[the remark this answers is not in the document]
Comment (PiDelport) [reply web-client/src/app/state/session-algorand.service.ts]: Done!

**Comment by PiDelport** [comment]
@longtomjr: Any comments on the updates, before we merge?

**Comment by longtomjr** [comment]
@PiDelport No comments on the changes. Looks good!

**Pull request merged**